src/components/MovieBanner.tsx

```diff
@@ -1,26 +1,14 @@
-import { useEffect, useState } from 'react';
-
 import Container from './Container';
 import PopularMovie from './PopularMovie';
 import { Movie } from '../models/index';
 
-import api from '../service/api';
-
 import '../styles/MovieBanner.scss';
 
-const MovieBanner = () => {
-  const [popularMovies, setPopularMovies] = useState<Movie[]>([]);
-
-  const getPopularMovies = (): void => {
-    api.getPopularQueryList().then(data => {
-      setPopularMovies(data);
-    });
-  };
-
-  useEffect(() => {
-    getPopularMovies();
-  }, []);
+type PopularMovies = {
+  popularMovies: Array<Movie>;
+};
 
+const MovieBanner = ({ popularMovies }: PopularMovies) => {
   const randomMovie =
     popularMovies[Math.floor(Math.random() * popularMovies.length)];
```

Comment:
Not really nice
This line fail the tests  - on first render popular movies are undefined
Also this line caused #220 since it give new random movie on each parent re-render

Reply:
I thought I have fixed the ad problem of the popular movie was being undefined, but this issue still appears during the testing. Locally, it seems that Movie Banner receives an array of objects (popular movies) during the first render. I am also not sure I got an issue that was mentioned in #220 request. Could you please explain it if I still have not to fix it yet?

Reply:
ok, Home component is rendered, it render MovieBanner and pass popularMovies as undefined, after render it call use effect - to fetch data, once relieved data, Home is being re-rendered and pass data to MovieBanner. 
But Home has few async calls - this means it will be re-rendered few times. And if MovieBanner has randomMovie as single variable it will be recalculated on each re-render - and give new random value. You can console.log this and check how many times it is logged and what is the value.
Solution - use useMemo or useState for randomMovie

Reply:
You seems to fix this, but the tests are failing because  Home component is rendered, it render MovieBanner and pass popularMovies as undefined
Add check to popular movies length inside useEffect

Reply:
Thanks a lot for a detailed explanation. I have created a state for a randomMovie and in useEffect (in MovieBanner component) i check  popularMovies.length, and if it's not empty we change a state of randomMovie